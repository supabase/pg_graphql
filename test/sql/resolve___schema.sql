begin;

    create table account(
        id serial primary key,
        email varchar(255) not null,
        encrypted_password varchar(255) not null,
        created_at timestamp not null,
        updated_at timestamp not null
    );


    create table blog(
        id serial primary key,
        owner_id integer not null references account(id),
        name varchar(255) not null,
        description varchar(255),
        created_at timestamp not null,
        updated_at timestamp not null
    );


    create type blog_post_status as enum ('PENDING', 'RELEASED');


    create table blog_post(
        id uuid not null default uuid_generate_v4() primary key,
        blog_id integer not null references blog(id),
        title varchar(255) not null,
        body varchar(10000),
        status blog_post_status not null,
        created_at timestamp not null,
        updated_at timestamp not null
    );


    select jsonb_pretty(
        gql.resolve($$
            query IntrospectionQuery {
              __schema {
                queryType {
                  name
                }
                mutationType {
                  name
                }
                types {
                  kind
                  name
                }
                directives {
                  name
                  description
                  locations
                }
              }
            }
        $$)
    );


rollback;
