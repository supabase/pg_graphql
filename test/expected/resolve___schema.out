begin;
    create table account(
        id serial primary key,
        email varchar(255) not null,
        encrypted_password varchar(255) not null,
        created_at timestamp not null,
        updated_at timestamp not null
    );
    create table blog(
        id serial primary key,
        owner_id integer not null references account(id),
        name varchar(255) not null,
        description varchar(255),
        created_at timestamp not null,
        updated_at timestamp not null
    );
    create type blog_post_status as enum ('PENDING', 'RELEASED');
    create table blog_post(
        id uuid not null default uuid_generate_v4() primary key,
        blog_id integer not null references blog(id),
        title varchar(255) not null,
        body varchar(10000),
        status blog_post_status not null,
        created_at timestamp not null,
        updated_at timestamp not null
    );
    select jsonb_pretty(
        gql.resolve($$
            query IntrospectionQuery {
              __schema {
                queryType {
                  name
                }
                mutationType {
                  name
                }
                types {
                  kind
                  name
                }
                directives {
                  name
                  description
                  locations
                }
              }
            }
        $$)
    );
                   jsonb_pretty                    
---------------------------------------------------
 {                                                +
     "data": {                                    +
         "__schema": {                            +
             "types": [                           +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "Account"            +
                 },                               +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "AccountConnection"  +
                 },                               +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "AccountEdge"        +
                 },                               +
                 {                                +
                     "kind": "SCALAR",            +
                     "name": "BigInt"             +
                 },                               +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "Blog"               +
                 },                               +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "BlogConnection"     +
                 },                               +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "BlogEdge"           +
                 },                               +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "BlogPost"           +
                 },                               +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "BlogPostConnection" +
                 },                               +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "BlogPostEdge"       +
                 },                               +
                 {                                +
                     "kind": "ENUM",              +
                     "name": "BlogPostStatus"     +
                 },                               +
                 {                                +
                     "kind": "SCALAR",            +
                     "name": "Boolean"            +
                 },                               +
                 {                                +
                     "kind": "SCALAR",            +
                     "name": "Cursor"             +
                 },                               +
                 {                                +
                     "kind": "SCALAR",            +
                     "name": "DateTime"           +
                 },                               +
                 {                                +
                     "kind": "SCALAR",            +
                     "name": "Float"              +
                 },                               +
                 {                                +
                     "kind": "SCALAR",            +
                     "name": "ID"                 +
                 },                               +
                 {                                +
                     "kind": "SCALAR",            +
                     "name": "Int"                +
                 },                               +
                 {                                +
                     "kind": "SCALAR",            +
                     "name": "JSON"               +
                 },                               +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "PageInfo"           +
                 },                               +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "Query"              +
                 },                               +
                 {                                +
                     "kind": "SCALAR",            +
                     "name": "String"             +
                 },                               +
                 {                                +
                     "kind": "SCALAR",            +
                     "name": "UUID"               +
                 },                               +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "__Directive"        +
                 },                               +
                 {                                +
                     "kind": "ENUM",              +
                     "name": "__DirectiveLocation"+
                 },                               +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "__EnumValue"        +
                 },                               +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "__Field"            +
                 },                               +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "__InputValue"       +
                 },                               +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "__Schema"           +
                 },                               +
                 {                                +
                     "kind": "OBJECT",            +
                     "name": "__Type"             +
                 },                               +
                 {                                +
                     "kind": "ENUM",              +
                     "name": "__TypeKind"         +
                 }                                +
             ],                                   +
             "queryType": {                       +
                 "name": "Query"                  +
             },                                   +
             "directives": [                      +
             ],                                   +
             "mutationType": null                 +
         }                                        +
     },                                           +
     "errors": [                                  +
     ]                                            +
 }
(1 row)

rollback;
